lld/ELF/Relocations.cpp

```diff
@@ -2367,7 +2367,63 @@ void elf::hexagonTLSSymbolUpdate(ArrayRef<OutputSection *> outputSections) {
       });
 }
 
+static bool matchesRefTo(const NoCrossRefCommand &cmd, StringRef osec) {
+  if (cmd.toFirst)
+    return cmd.outputSections[0] == osec;
+  return llvm::is_contained(cmd.outputSections, osec);
+}
+
+template <class ELFT, class Rels>
+static void scanCrossRefs(const NoCrossRefCommand &cmd, OutputSection *osec,
+                          InputSection *sec, Rels rels) {
```

Comment:
For a NoCrossRefCommand cmd with `toFirst == true` can we early exit if osec == cmd.outputSections[0]? 

This would be the case where we search the relocations from `tosection`.

Reply:
Added an early return to `elf::checkNoCrossRefs`: `(noxref.toFirst && noxref.outputSections[0] == osec->name)`